pkg/build/generator/generator.go

```diff
@@ -293,6 +294,8 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	build.Annotations = make(map[string]string)
+	build.Annotations[buildapi.BuildNumberAnnotation] = strconv.Itoa(bc.Status.LastVersion)
```

Comment:
should we copy the annotations from the cloned build? is it weird for two builds to have the same number?